sympy/functions/elementary/piecewise.py

```diff
@@ -226,6 +226,12 @@ def _eval_as_leading_term(self, x, logx, cdir):
         for e, c in self.args:
             if c == True or c.subs(x, 0) == True:
                 return e.as_leading_term(x)
+            elif cdir == -1 and c.subs(x, 0) == False:
+                if c.func == Lt and c.lhs==x:
+                    return e.as_leading_term(x)
+            elif cdir == 1 and c.subs(x, 0) == False:
+                if c.func == Gt and c.lhs==x:
+                    return e.as_leading_term(x)
```

Comment:
This is the right sort of thing to fix this but I think that the logic needs to be handled more carefully. The first case substituting into the condition is still incorrect: we can't check `c.subs(x, 0)` like that. This case fails for example:
```python
In [1]: p = Piecewise((1, Ne(x, 0)), (0, True))

In [2]: p
Out[2]: 
⎧1  for x ≠ 0
⎨            
⎩0  otherwise

In [3]: p.limit(x, 0)
Out[3]: 0

In [4]: p.limit(x, 0, '-')
Out[4]: 0

In [5]: p.limit(x, 0, '+')
Out[5]: 0

In [6]: p.limit(x, 0, '+-')
Out[6]: 0
```
All limits in this case should be `1`. The inverted case `Piecewise((0, Eq(x, 0)), (1, True))` also fails.

A function is needed that can determine if the condition applies for a left, right or both-sided limit and the function will need to return three cases: yes, no and maybe.

I think that `subs` should not be used at all because the function does not need to be defined at the limit point and even if it is the value there can be unrelated to the values on either side. None of the limits depends on the actual value of the function evaluated at the limit point.

Reply:
Is there a way to evaluate functions from a particular infinitesimal difference?
Maybe based on cdir we can evaluate
c.subs(x, 0 + simpy.core.numbers.VerySmallNumber) or c.subs(x, 0 - simpy.core.numbers.VerySmallNumber)
and use that result?

Reply:
There isn't a way to do that. It might make sense as an idea but ultimately the code to make it work would not be much different from the code that would determine which condition applies here.